### PR TITLE
Removed Slic internal data members 

### PR DESCRIPTION
**Pull request by bentoi**
This PR gets rid of internal data members which weren't properties. The data members are now private. The SlicSocket Mutex  remains internal however because it's needed by the SlicStream implementation.

### CODE REVIEW

File: csharp/src/Ice/SlicStream.cs [hunk 1]
```diff
@@ -318,44 +281,21 @@ internal override void ReceivedReset(long errorCode)
...
-                    if (IsBidirectional)
-                    {
-                        Interlocked.Decrement(ref _socket.BidirectionalStreamCount);
-                    }
-                    else
-                    {
-                        Interlocked.Decrement(ref _socket.UnidirectionalStreamCount);
-                    }
-
                     if (notifyPeer)
```
Comment (pepone) [inline csharp/src/Ice/SlicStream.cs:292]: can we replace the double if by and condition now?
```
if (IsIncoming && notifyPeer){ ...}
```

File: csharp/src/Ice/SlicStream.cs [hunk 2]
```diff
@@ -318,44 +281,21 @@ internal override void ReceivedReset(long errorCode)
 
         private bool ReleaseFlowControlCredit(bool notifyPeer = false)
         {
-            if (IsControl)
-            {
-                return true;
-            }
-
             // If the flow control credit is not already released, decreases the bidirectional or unidirectional
             // semaphore (for outgoing streams) / count (for incoming streams).
```
Comment (pepone) [inline csharp/src/Ice/SlicStream.cs:285]: I think we should update the comment as this no longer use the semaphore directly

File: csharp/src/Ice/SocketStream.cs [hunk 3]
```diff
@@ -52,7 +52,9 @@ public long Id
         /// <summary>The Reset event is triggered when a reset frame is received.</summary>
         internal event Action<long>? Reset;
 
-        private protected bool IsStarted => _id != -1;
+        /// <summary>Returns true if the stream ID is assigned</summary>
+        internal bool IsStarted => _id != -1;
```
Comment (bernardnormier) [inline csharp/src/Ice/SocketStream.cs:56]: `_id` is not volatile or protected by a mutex - would be nice to add a comment explaining why.

**Review by bernardnormier — APPROVED** [review]
Looks fine. I would still prefer to see the mutex private, just as a general coding practice.

**Review by pepone — APPROVED** [review]
Looks good

File: csharp/src/Ice/SocketStream.cs (hunk 3 above)
Comment (bentoi) [reply csharp/src/Ice/SocketStream.cs:56]: Added the following:
```
        // Depending on the stream implementation, the _id can be assigned on construction or only once SendAsync
        // is called. Once it's assigned, it's immutable. The specialization of the stream is responsible for not
        // accessing this data member concurrently when it's not safe.
```

**Comment by bentoi** [comment]
I've pushed another fix to get rid of the internal Mutex by delegating part of the packet send to the socket. Can you check again?

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: csharp/src/Ice/SlicStream.cs (hunk 1 above)
Comment (bentoi) [reply csharp/src/Ice/SlicStream.cs:278]: Yes, fixed!

**Pull request merged**